plans/i1.md

```diff
@@ -0,0 +1,8 @@
+# Iteration 1 (July 2019)
+
+- Continue understanding how lack of "type extension" (e.g. `<table is="bx-table">`) affects our library and how we can get around it
+- Tests to cover most components
+- Start developing some key components
+  - Form components (Checkbox/Number input/Text input/Text input)
+  - Combo box/Multi select
+  - Data table
```

Comment:
Some of the problems with lack of extensions here can be sorted with some extra directives + a reasonable table model ... carbon angular for example, has a well defined model for table interactions (https://github.com/IBM/carbon-components-angular/blob/master/src/table/table-model.class.ts) ... something along those lines could help with our lack of native element extension

Reply:
Thank you for sharing the example! Your link is interesting for state management capability, given we need to provide at least a guidance there. Another thing we need to look into is if we can represent each table headers/rows/cells as components.